### DISCUSSION TITLE
[QUESTION] Longhorn plugin failed to encrypt device

### DISCUSSION THREAD

**Discussion by ipapatha85** [longhorn/longhorn#6780] [Q&A]
## Question

Hi!

We have followed the [instructions ](https://longhorn.io/docs/1.4.0/advanced-resources/security/volume-encryption/) to use an encrypted volume. However, cryptsetup fails to perform any actions with exit code 127. Cryptsetup is installed on already running plugin container with `apt-get install -y cryptsetup`.

Do we miss something else?

## Environment

 - Longhorn version: v.1.2.4
 - Kubernetes version: v.1.23.6
 
## Log

```
time="2023-02-24T12:43:34Z" level=debug msg="volume pvc-fed678e3-6851-4cea-8a80-ace8b58e321f device /dev/longhorn/pvc-fed678e3-6851-4cea-8a80-ace8b58e321f contains filesystem of format "
time="2023-02-24T12:43:34Z" level=debug msg="Encrypting device /dev/longhorn/pvc-fed678e3-6851-4cea-8a80-ace8b58e321f with LUKS"
time="2023-02-24T12:43:34Z" level=error msg="NodeStageVolume: err: rpc error: code = Internal desc = failed to encrypt device /dev/longhorn/pvc-fed678e3-6851-4cea-8a80-ace8b58e321f with LUKS: failed to run cryptsetup args: [-q luksFormat --type luks2 --hash sha256 /dev/longhorn/pvc-fed678e3-6851-4cea-8a80-ace8b58e321f -d /dev/stdin] output: error: exit status 127"
```

**Comment by mantissahz** [comment]
Hi, @ipapatha85 
Could you check if `cryptsetup` was installed on your `worker nodes` according to the doc [Requirement](https://longhorn.io/docs/1.2.6/advanced-resources/volume-encryption/#requirements)?
Thank you.

**Comment by ipapatha85** [comment]
it worked!

I installed it on the container although, as you highlighted, this shall be done on the worker node.

many thanks!
